sdk/dotnet/Pulumi/Core/OutputExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright 2016-2019, Pulumi Corporation
+
+using System.Collections.Immutable;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// Extension methods for <see cref="Output{T}"/>.
+    /// </summary>
+    public static class OutputExtensions
+    {
+        /// <summary>
+        /// Convert an output containing an array to an output containing the array element
+        /// at the specified index.
+        /// </summary>
+        /// <typeparam name="T">The type of elements in the array.</typeparam>
+        /// <param name="array">An array wrapped into <see cref="Output{T}"/>.</param>
+        /// <param name="index">An index to get an element at.</param>
+        /// <returns>An <see cref="Output{T}"/> containing an array element.</returns>
+        public static Output<T> GetAt<T>(this Output<ImmutableArray<T>> array, int index)
```

Comment:
this is actually a really interesting idea, and perhaps something we can do for specific packages as well.  for example, if you pull in Pulumi.Aws.S3 you'd get all the extensions to lift properties on an `Output<SomeS3Type>`.

Reply:
Yeah, we could...
So, are you good with this API for output arrays? (this PR)

Reply:
Almost.  We shoudl allow the index to be an `Input<int>`.  And we should also have a `.Length()` extension that produces an `Output<int>` imo.

Reply:
Added

Reply:
thanks!